### ISSUE TITLE
Give different weights to different training samples

### ISSUE THREAD

**Issue by rispoli997** [UKPLab/sentence-transformers#2871]
Hello, I have to train a ST model on a very specific task, unfortunately my dataset is not big enough, therefore I was thinking to augment my data using an LLM. As my task is quite specific and demands high-precision, the synthetic data supplied by the LLM, while being good to give the model a "basic knowledge about the data", being too general, might hurt the performances of the model when it has to be very precise. As such I was thinking of assigning a lower weight to synthetic samples, and an higher one to the samples that belong to the real dataset. Is there any built-in way to do this? At the moment I am using the MNR loss.

**Comment by ir2718** [comment]
Hi,

There is no built-in way to do this, but can easily be hacked. Make sure your dataset has a different label for the LLM examples, eg. 1 for the original examples, 2 for the LLM examples.

```
import torch
import torch.nn as nn
from typing import Iterable
from torch import Tensor


class LLMWeightedMultipleNegativesRankingLoss(MultipleNegativesRankingLoss):

    def __init__(self, llm_weight=0.3, llm_label=2, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.llm_weight = llm_weight
        self.llm_label = llm_label
        self.cross_entropy_loss = nn.CrossEntropyLoss(reduction="none")

    def forward(self, sentence_features: Iterable[dict[str, Tensor]], labels: Tensor) -> Tensor:
        reps = [self.model(sentence_feature)["sentence_embedding"] for sentence_feature in sentence_features]
        embeddings_a = reps[0]
        embeddings_b = torch.cat(reps[1:])

        scores = self.similarity_fct(embeddings_a, embeddings_b) * self.scale
        # Example a[i] should match with b[i]
        range_labels = torch.arange(0, scores.size(0), device=scores.device)

        # first calculate the loss without the reduction
        loss_wo_reduction = self.cross_entropy_loss(scores, range_labels)

        # weigh the original examples with 1
        default_loss_sum = loss_wo_reduction[labels!=self.llm_label].sum()

        # weigh the llm examples with self.llm_weight
        llm_weighted_loss_sum = (loss_wo_reduction[labels==self.llm_label] * self.llm_weight).sum()

        # combine the two into a mean reduction
        mean_loss = (default_loss_sum + llm_weighted_loss_sum) / loss_wo_reduction.size(0)
        return mean_loss    


train_loss = LLMWeightedMultipleNegativesRankingLoss(0.3, 2, model)
```
